src/dev/build/tasks/create_cdn_assets_task.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { readFileSync } from 'fs';
+import { access } from 'fs/promises';
+
+import { resolve, dirname } from 'path';
+import { asyncForEach } from '@kbn/std';
+import { Jsonc } from '@kbn/repo-packages';
+
+import del from 'del';
+import globby from 'globby';
+
+import { mkdirp, compressTar, Task, copyAll } from '../lib';
+
+export const CreateCdnAssets: Task = {
+  description: 'Creating CDN assets',
+
+  async run(config, log, build) {
+    const buildSource = build.resolvePath();
+    const buildNum = config.getBuildNumber();
+    const buildVersion = config.getBuildVersion();
+    const assets = config.resolveFromRepo('build', 'cdn-assets');
+    const bundles = resolve(assets, String(buildNum), 'bundles');
+    const plugin = resolve(bundles, 'plugin');
+
+    await del(assets);
+    await mkdirp(assets);
+
+    // Plugins
+    await mkdirp(plugin);
+    const plugins = globby.sync([`${buildSource}/node_modules/@kbn/**/*/kibana.jsonc`]);
+    await asyncForEach(plugins, async (path) => {
+      const manifest = Jsonc.parse(readFileSync(path, 'utf8')) as any;
+      if (manifest?.plugin?.id) {
+        const pluginRoot = resolve(dirname(path));
+
+        try {
+          const assetsSource = resolve(pluginRoot, 'assets');
+          const assetsDest = resolve(plugin, manifest.plugin.id, 'assets');
+          await access(assetsSource);
+          await copyAll(assetsSource, assetsDest);
```

Comment:
If I read the path correctly, then I think this is wrong. 

The plugins static assets aren't accessed/exposed as other bundles via `/assets/{buildNum}/bundles/plugin/{id}/assets/{asset_path}`, but via `/plugins/{id}/assets/{asset_path}`

E.g

https://github.com/elastic/kibana/blob/497e6c36b19ea9103fe218dab919269b87459352/src/plugins/home/server/services/sample_data/data_sets/flights/index.ts#L27

For this file: `src/plugins/home/public/assets/sample_data_resources/flights/dashboard.webp`

Reply:
I pushed https://github.com/elastic/kibana/pull/169707/commits/06a3ded566da314948484e5684aab3491b4e55a3